indico/modules/events/layout/models/menu.py

```diff
@@ -125,6 +125,8 @@ def is_orphaned(self):
     def is_visible(self):
         if not self.is_enabled:
             return False
+        if self.registered_only and not self.event.is_user_registered(session.user):
+            return False
```

Comment:
We now have access checks both in here and in `MenuEntryData.visible`. Can you either move your check in there or the one from there in here:

```
        if self.hide_if_restricted and not event.can_access(session.user):
            return False
```

If you keep the checks here IIRC you are supposed to use `self.event_ref` instead of `self.event`.